apis/v1beta1/types.go

```diff
@@ -25,6 +25,7 @@ import (
 // A ProviderConfigSpec defines the desired state of a Provider.
 type ProviderConfigSpec struct {
 	xpv1.ProviderConfigSpec `json:",inline"`
+	GoogleAppCreds          xpv1.ProviderCredentials `json:"google-app-creds,omitempty"`
```

Comment:
Would it make sense for this to be a new kind of `source` within the existing `credentials` field? Something like `source: GoogleAppSecret`? This would require duplicating rather than embedding most of the `xpv1.ProviderConfigSpec`, but doing so is totally fine.

Reply:
I've created https://github.com/crossplane/crossplane-runtime/issues/281 to look at a new CredentialSource specific to Google Cloud.